packages/xdl/src/modules/config.js

```diff
@@ -551,6 +551,20 @@ const universalModules = [
   },
 ];
 
+const vendoredNativeModules = [
+  { libName: '@expo/vector-icons', sdkVersions: '>=26.0.0', isNativeModule: true },
+  { libName: '@react-native-community/netinfo', sdkVersions: '>=33.0.0', isNativeModule: true },
+  { libName: 'lottie-react-native', sdkVersions: '>=26.0.0', isNativeModule: true },
+  { libName: 'react-native-branch', sdkVersions: '>=26.0.0', isNativeModule: true },
+  { libName: 'react-native-gesture-handler', sdkVersions: '>=26.0.0', isNativeModule: true },
+  { libName: 'react-native-maps', sdkVersions: '>=26.0.0', isNativeModule: true },
+  { libName: 'react-native-reanimated', sdkVersions: '>=28.0.0', isNativeModule: true },
+  { libName: 'react-native-screens', sdkVersions: '>=30.0.0', isNativeModule: true },
+  { libName: 'react-native-svg', sdkVersions: '>=26.0.0', isNativeModule: true },
+  { libName: 'react-native-view-shot', sdkVersions: '>=26.0.0', isNativeModule: true },
+  { libName: 'react-native-webview', sdkVersions: '>=33.0.0', isNativeModule: true },
```

Comment:
@brentvatne could you verify that this list is correct? I compiled it manually by comparing the dependencies of the `expo` package and the gulp tasks like [this one](https://github.com/expo/expo/blob/5f93ff3156234e0bf8010abf42256cdd581e9796/tools/gulpfile.js#L121-L134) in the expo/expo repo.

There are `updateVendoredNativeModule` tasks for `amazon-cognito-identity-js`, `react-tipsi-stripe`, `react-native-admob` and `react-native-fbads` too, but I didn't find them in the SDK reference and couldn't figure out if they should be installable in managed apps or not.

Reply:
good point - i think no need to add those ones.

ultimately i don't think this should live in xdl because it requires that we publish a new version of xdl and expo-cli any time we want to change a version. it'd probably be best if it lived at some remote endpoint, like native.directory or some endpoint on www.

Reply:
> ultimately i don't think this should live in xdl because it requires that we publish a new version of xdl and expo-cli any time we want to change a version.
I 100% agree this data shouldn't live in XDL, I'm only adding it here to support older SDK versions which won't have this data available in the `expo` package.

My idea is:
* For SDK <33, use the existing list of Unimodules in XDL combined with version numbers from `expo/package.json`.
* For SDK >=33, use a single file in the `expo` package that includes names of the supported packages and their versions. It's a finite set, because the only packages that are included are those whose native code is baked in the SDK and Expo Client.

I'd like to see how far we can get with just a flat file alongside the code that's distributed for each SDK version, without relying on a special endpoint that we'd host for this functionality. I think this has a few nice properties:
- Versioning is simple. The data is versioned in git together with the SDK that it describes and it's easy to find the history of changes and compare versions. 
- Making changes to it is not different from contributing any other changes to the SDK. You can commit / send a PR. Anyone who has permissions to publish `expo` to npm can make a release without a need to have separate write permissions to our internal API (or S3) or deployment permissions to www.
- The data is available offline in managed Expo projects, without any special caching considerations on our part. (Together with npm/Yarn cache it's possible to install packages you've previously downloaded without having an internet connection.)
- To use this feature it's inevitable that you must be able to connect to the npm registry. Some people need to use a proxy for this. Not relying on an additional web service (when it's not necessary) makes it available to more people.

Reply:
ok that makes sense, let's see how far we can go with this approach then!